app/views/editor/thang/ThangTypeColorsTabView.coffee

```diff
@@ -122,31 +191,44 @@ module.exports = class ThangTypeColorsTabView extends CocoView
 
   # Attaches hard coded color tabs for manipulating defined color groups on the ThangType
   createColorGroupTintButtons: ->
+    return if @destroyed
+    return unless @tintAssignments
     buttons = $('<div></div>').prop('id', 'saved-color-tabs')
     buttons.append($("<h1>Saved Color Presets</h1>"))
-    hardCodedGroups = {
-      skin: ['#FFDBAC', '#F1C27D', '#E0AC69', '#C68642', '#8D5524'],
-      hair: ['#090808', '#3B3024', '#DEBC99', '#8D4A43', '#71635A', '#A7856A', '#7FB4BE', '#f9a1fc', '#236d21']
-    }
 
-    for groupColor, colors of hardCodedGroups
-      @addColorTintGroup(buttons, groupColor, colors)
+    colors = @tintAssignments.data
+    for tint, i in colors
+      tintName = tint.name
+      @addColorTintGroup(buttons, tintName, tint.allowedTints or [], i)
 
     @$el.find('#saved-color-tabs').replaceWith(buttons)
 
-  addColorTintGroup: (buttons, groupName, colors) ->
-    buttons.append($("<h3>#{groupName}</h3>"))
+  addColorTintGroup: (buttons, tintName, tints, index) ->
+    buttons.append($("<h3>#{tintName}</h3>"))
+    saveButton = $("<button>#{tintName}</button>")
+    buttons.append($('<button />', {
+      text: "Save '#{tintName}' Tints",
+      class: 'save-btn',
+      click: ((index) => () =>
```

Comment:
This is a way to add a closure in coffeescript. Let's me create a save button for each tint index.

Reply:
This is a great candidate for a comment in code :)